src/components/SearchCard.js

```diff
@@ -0,0 +1,215 @@
+import throttle from "lodash/throttle";
+import React, { useMemo, useState } from "react";
+import { constants } from "../util/helpers";
+const { Input, Card, Button, Result, Select } = require("antd");
+const { Option } = Select;
+const defaultNetwork = constants.CONTRACT_NETWORKS[0];
+
+export const SearchCard = ({ session }) => {
+  const [loading, setLoading] = useState(false);
+  const [installing, setInstalling] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [contract, setContract] = useState("");
+  const [error, setError] = useState("");
+  const [defs, setDefs] = useState([]);
+  const [network, setNetwork] = useState(defaultNetwork.value);
+
+  const resetState = () => {
+    setLoading(false);
+    setSuccess(false);
+    setInstalling(false);
+    setContract("");
+    setDefs([]);
+  };
+
+  function onChange(value) {
+    setNetwork(value);
+  }
+
+  // TEMPORARY FUNCTION TO REMOVE FUNCTIONS WITH ZERO LENGTH PARAM NAMES
+  // SEE: https://github.com/GridPlus/gridplus-sdk/issues/1540xfDcC959b0AA82E288E4154cB1C770C6c4e958a91
+  function TMP_REMOVE_ZERO_LEN_PARAMS(defs) {
+    const newDefs = [];
+    defs.forEach((def) => {
+      let shouldAdd = true;
+      if (def.name.length === 0) {
+        shouldAdd = false;
+      } else {
+        def.params.forEach((param) => {
+          if (param.name.length === 0) shouldAdd = false;
+        });
+      }
+      if (shouldAdd === true) newDefs.push(def);
+    });
+    return newDefs;
+  }
+
+  const getNetworkApiUrl = () =>
+    constants.CONTRACT_NETWORKS.find((cn) => cn.value === network).api;
+
+  function fetchContractData(input) {
+    if (
+      input.slice(0, 2) !== "0x" ||
+      false === /^[0-9a-fA-F]+$/.test(input.slice(2)) ||
+      input.length !== 42
+    ) {
+      // Not a valid address
+      setError("Invalid Ethereum contract address");
+      resetState();
+    } else {
+      fetch(`${getNetworkApiUrl()}${input}`)
+        .then((response) => response.json())
+        .then((resp) => {
+          // Map confusing error strings to better descriptions
+          if (resp.result === "Contract source code not verified") {
+            resp.result =
+              "Contract source code not published to Etherscan or not verified. Cannot determine data.";
```

Comment:
You should add the name of the explorer ("Etherscan", "Polygonscan", etc) to `CONTRACT_NETWORKS` and replace `Etherscan` here with that variable.